### PR TITLE
Add 4 more mechanics and refactor a bit

### PR DESCRIPTION
**Pull request by Nailec**

### CODE REVIEW

File: showdown/engine/instruction_generator.py [hunk 1]
```diff
@@ -77,27 +77,18 @@ def can_trick_items(attacker, defender):
...
-        return False
-
-    elif ('arceus' in attacker.id or 'arceus' in defender.id) and (attacker_item.endswith('plate') or defender_item.endswith('plate')):
-        return False
-
-    elif ('genesect' in attacker.id or 'genesect' in defender.id) and (attacker_item.endswith('drive') or defender_item.endswith('drive')):
-        return False
-
-    return True
+    # item is trickable if none of the following wonditions is matched
```
Comment (pmariglia) [inline showdown/engine/instruction_generator.py:80]:
```suggestion
```

IMO just a noisy comment, the function name describes what this does

File: showdown/engine/special_effects/abilities/on_switch_in.py [hunk 2]
```diff
@@ -90,25 +90,51 @@ def mistysurge(state, attacking_side, attacking_pokemon, defending_side, defendi
...
+            min(6-defending_pokemon.special_attack_boost, 2) #stop boosting when it reaches 6
         )]
-    return None
+
+    return [(
+        constants.MUTATOR_UNBOOST,
+        defending_side,
+        constants.ATTACK,
+        min(1, 6+defending_pokemon.attack_boost)
+    )]
```
Comment (pmariglia) [inline showdown/engine/special_effects/abilities/on_switch_in.py:137]: The removal of the -6 attack_boost check is causing a test to fail.

It can be added back after the ability-checks here:

```suggestion
    if defending_pokemon.attack_boost > -6:
        return [(
            constants.MUTATOR_UNBOOST,
            defending_side,
            constants.ATTACK,
            min(1, 6+defending_pokemon.attack_boost)
        )]
```

File: showdown/engine/special_effects/abilities/on_switch_in.py [hunk 3]
```diff
@@ -90,25 +90,51 @@ def mistysurge(state, attacking_side, attacking_pokemon, defending_side, defendi
...
-            constants.MUTATOR_BOOST,
+            constants.MUTATOR_UNBOOST,
             defending_side,
             constants.ATTACK,
             1
+        ), (
+            constants.MUTATOR_BOOST,
+            defending_side,
+            constants.SPECIAL_ATTACK,
+            min(6-defending_pokemon.special_attack_boost, 2) #stop boosting when it reaches 6
```
Comment (pmariglia) [inline showdown/engine/special_effects/abilities/on_switch_in.py:129]: The micro-est of micro-optimizations:

Could these three conditionals be grouped into if/elif/elif instead of if/if/if ?

**Comment by pmariglia** [comment]
Looks good. Yeah I see what you mean about the edge-cases for triggering all of those ability effects. Gets really spaghetti really fast.

This is fine, except for the failing test.
